sql/core/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -464,6 +464,16 @@ object SQLConf {
     .intConf
     .createWithDefault(20)
 
+  val DATA_SOURCE_TABLE_RELATION_CACHE_MAX_SIZE =
+    SQLConfigBuilder("spark.sql.sources.tableRelationCache.maxSize")
```

Comment:
We are having the similar cache size conf. 
```Scala
  val HIVE_FILESOURCE_PARTITION_FILE_CACHE_SIZE =
    SQLConfigBuilder("spark.sql.hive.filesourcePartitionFileCacheSize")
```

Maybe you can change it to make the naming more consistent.